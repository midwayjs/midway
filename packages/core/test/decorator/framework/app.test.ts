import {
  App,
  getClassMetadata,
  INJECT_CUSTOM_PROPERTY,
  ApplicationContext,
} from '../../../src';

class Test {
  @App()
  hhh: any;

  @ApplicationContext()
  container: any;
}

describe('/test/framework/app.test.ts', () => {
  it('app decorator should be ok', () => {
    let data = getClassMetadata(INJECT_CUSTOM_PROPERTY, Test);
    expect(data).toMatchSnapshot();
  });
});
