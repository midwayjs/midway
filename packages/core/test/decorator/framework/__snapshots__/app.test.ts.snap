// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`/test/framework/app.test.ts app decorator should be ok 1`] = `
{
  "container": {
    "impl": true,
    "key": "__midway_application_context__",
    "metadata": {},
    "propertyName": "container",
  },
  "hhh": {
    "impl": true,
    "key": "__midway_framework_app__",
    "metadata": {
      "type": undefined,
    },
    "propertyName": "hhh",
  },
}
`;
